c-data-channels/bridge.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+/*
+#include <stddef.h>
+typedef struct GoDataChannelMessage
+{
+    // bool in cgo is really weird, so it's simpler to just use classic int as bool
+    int is_string;
+    void *data;
+    size_t data_len;
+} GoDataChannelMessage;
+
+typedef int GoDataChannel;
+typedef void (*GoOnDataChannelFunc)(GoDataChannel);
+typedef void (*GoOnOpenFunc)(GoDataChannel);
+typedef void (*GoOnMessageFunc)(GoDataChannel, GoDataChannelMessage);
+
+// Calling C function pointers is currently not supported, however you can
+// declare Go variables which hold C function pointers and pass them back
+// and forth between Go and C. C code may call function pointers received from Go.
+// Reference: https://golang.org/cmd/cgo/#hdr-Go_references_to_C
+inline void bridge_on_data_channel(GoOnDataChannelFunc cb, GoDataChannel d)
+{
+    cb(d);
+}
+
+inline void bridge_on_open(GoOnOpenFunc cb, GoDataChannel d)
+{
+	cb(d);
+}
+
+inline void bridge_on_message(GoOnMessageFunc cb, GoDataChannel d, GoDataChannelMessage msg)
+{
+	cb(d, msg);
+}
+*/
+import "C"
+
+import (
+	"github.com/pion/webrtc/v2"
+)
+
+/*
```

Comment:
I think it would be a lot simpler to just store DataChannels by label in a map. 

Then when you send you can just call `label`, `message`. Also mind adding a simple C function to generate a pseudo-random string?

Just put a bunch of acceptable characters in a `char charset[]` and then pick them out via `rand()`